octodns/provider/route53.py

```diff
@@ -704,14 +1118,91 @@ def _mod_Update(self, change):
             self._gen_mods('CREATE', creates) + \
             self._gen_mods('UPSERT', upserts)
 
-    def _mod_Delete(self, change):
+    def _mod_Delete(self, change, zone_id):
         # Existing is the thing that needs to be deleted
-        existing_records = self._gen_records(change.existing, creating=False)
+        existing_records = self._gen_records(change.existing, zone_id,
+                                             creating=False)
         # Now is a good time to clear out all the health checks since we know
         # we're done with them
         self._gc_health_checks(change.existing, [])
         return self._gen_mods('DELETE', existing_records)
 
```

Comment:
The next few functions just break up and rework _extra_changes to support dynamic records. The code is pretty different, but the before and after logic is identical for geo and just adds support for dynamic.